src/nvim/normal.c

```diff
@@ -635,6 +635,21 @@ static void normal_redraw_mode_message(NormalState *s)
   emsg_on_display = false;
 }
 
+static int add_vgetc(void)
+{
+   while(true) {
+     int key = state_vgetc("nx");
+     if (key == K_IGNORE) {
+       continue;
+     } else if (key == K_EVENT) {
+        state_process_events("nx");
+        continue;
+     }
+     return key;
+   }
+}
+
+
 // TODO(tarruda): Split into a "normal pending" state that can handle K_EVENT
 static void normal_get_additional_char(NormalState *s)
```

Comment:
The comment on line 653 suggests a different approach I think. Maybe it's more natural: add a new `foo_enter`, `foo_execute()`, etc. family of state-related functions.

Reply:
What would be the gain of changing the structure of the code so much? The purpose of `state_vgetc` and `state_process_events` is to easily support temporary states that don't handle mappings (on their own, "normal additional" can of course still process rhs of "normal normal" mapping) and really need no "internal" state flag, but we still want to announce it in `nvim_get_mode' (either, as I do by overriding `mode(1)` or alternatively by adding third field)

And I'm looking forward to seeing the multicursor branch :)

Reply:
can the TODO comment be deleted then? :)